lib/surface/compiler.ex

```diff
@@ -414,6 +418,26 @@ defmodule Surface.Compiler do
     end)
   end
 
+  defp attribute_value_as_ast(attributes, attr_name, default, meta) do
```

Comment:
We should probably have a place to centralize this kind of functions. Maybe `AST.Attribute`? I don't like the name `value_as_ast` but using just `value` is confusing. I'd love to have a nice name that could represent `AST.Text | AST. AttributeExpr` but I don't :)

Reply:
Yeah - there are a couple of similar functions spread throughout the code for operating on the ast that I think would also fall into a similar category. I think moving these into the AST module would make sense, and I think we'll continue finding more of these types of things as we do more work with the AST